File: src/lua_data.cpp
#include <sstream>
#include <iostream>

#include "cpp_tolua/lua_data.h"

using namespace std;

bool LuaData::operator<(const LuaData& data) const
{
	if (m_i < data.m_i)
	{
		return true;
	}
	if (m_i == data.m_i)
	{
		if (m_s < data.m_s)
		{
			return true;
		}
		else
		{
			return false;
		}
	}
	return false;
}

string LuaData::toString() const
{
	stringstream ss;
	ss << "i:" << m_i << ",s:" << m_s << ",vecsize:" << m_vec.size() << "[";
	for (vector<int>::const_iterator it = m_vec.begin(); it != m_vec.end(); ++it)
	{
		ss << (*it) << ",";
	}
	ss << "],p.i:" << m_p.m_i << ",p.s" << m_p.m_s << ",pvecsize:" << m_p.m_vec.size() << "[";
	for (vector<int>::const_iterator it = m_p.m_vec.begin(); it != m_p.m_vec.end(); ++it)
	{
		ss << (*it) << ",";
	}
	ss << "]";
	return ss.str();
}

File: lua/lua_data.lua
function log_print(str)
    print("[lua]"..str)
end

log_print("lua_data 开始")

function TestLuaData(p)
    local d = ToLuaData(p)
    log_print(d:toString())

    d.m_i = d.m_i * 10
    d.m_s = d.m_s .. "_"
    v = ToLuaVectorInt(d.m_vec)
    v:push_back(7)

    d.m_p.m_i = d.m_p.m_i * 10
    d.m_p.m_s = d.m_p.m_s .. "_"
    dv = ToLuaVectorInt(d.m_p.m_vec)
    dv:push_back(77)

    log_print(d:toString())
end

log_print("lua_data 结束")
